lib/phonebook/class/complete_member.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:flutter/material.dart';
+import 'package:myecl/phonebook/class/membership.dart';
+import 'member.dart';
+
+class CompleteMember {
+  CompleteMember({
+    required this.member,
+    required this.memberships,
+  });
+
+  late final Member member;
+  late final List<Membership> memberships;
+
+  CompleteMember.fromJSON(Map<String, dynamic> json) {
+    debugPrint(json.toString());
+    member = Member(
+        name: json['name'],
+        firstname: json['firstname'],
+        nickname: json['nickname'],
+        id: json['id'],
+        email: json['email'],
+        promotion: json['promotion']??"");
+    memberships = List<Membership>.from(json['memberships']
+        .map((membership) => Membership.fromJSON(membership)));
+        
+  }
+
+  Map<String, dynamic> toJSON() {
+    final data = <String, dynamic>{
+      'member': member.id,
+      'memberships': memberships.map((e) => e.toJSON()).toList(),
+    };
+    return data;
+  }
+
+  CompleteMember copyWith({
+    Member? member,
+    List<Membership>? membership,
+  }) {
+    return CompleteMember(
+      member: member ?? this.member,
+      memberships: membership ?? memberships,
+    );
+  }
+
+  CompleteMember.empty() {
```

Comment:
Does this function really works, because member is final, so it should throw an AlreadyInitializedError